addons/OQ_Toolkit/OQ_ARVROrigin/scripts/OQ_ARVROrigin.gd

```diff
@@ -4,11 +4,15 @@ export var debug_information := false;
 
 var is_fixed = false;
 
+export(bool) var initialize_vr := true
+
 # Sets up everything as it is expected by the helper scripts in the vr singleton
 func _enter_tree():
 	if (!vr):
 		vr.log_error("in OQ_ARVROrigin._enter_tree(): no vr singleton");
 		return;
+	if (!vr.initialized and initialize_vr):
```

Comment:
I like the idea to make it easier to use by auto-initializing and not requiring it user code; but as you have written this will make some settings and other required code (like the scene switch root) slightly more intransparent for users who will rely on this mechanism.
Also in my applications I have a OQ_ARVROrigin in every scene I switch to; but this should not be a problem I think with the guard.

Would it be possible to make the option part of the plugin settings (somehting like auto_initialize_vr)?

Reply:
absolutely, and probably needs to be for flexibility.